pyro/contrib/tracking/distributions.py

```diff
@@ -37,14 +37,34 @@ def __init__(self, x0, P0, dynamic_model, measurement_cov, time_steps=1, dt=1.,
         self.dynamic_model = dynamic_model
         self.measurement_cov = measurement_cov
         self.dt = dt
+        assert not x0.shape[-1] % 2, 'position and velocity vectors must be the same dimension'
         batch_shape = x0.shape[:-1]
-        event_shape = (time_steps,) + x0.shape[-1:]
+        event_shape = (time_steps,) + (x0.shape[-1] // 2,)
         super(EKFDistribution, self).__init__(batch_shape, event_shape,
                                               validate_args=validate_args)
 
     def rsample(self, sample_shape=torch.Size()):
         raise NotImplementedError('TODO: implement forward filter backward sample')
 
+    def get_states(self, value):
```

Comment:
nit: I would rename this `filter_states()` or similar. `get_*()` suggests accessing an attribute, whereas you're actually performing computation here. Moreover you are only filtering forward, not filtering-forward-and-smoothing-backward, so it would be good to emphasize that you are returning filtered states rather than posterior states (only the final state will be a posterior marginal). In the future we could add a `filter_smooth_states()` that computes exact marginals at each time point.